### ISSUE TITLE
[Feature request] add support for throwing from postinit

### ISSUE THREAD

**Issue by lydia-duncan** [chapel-lang/chapel#21418]
It may be desirable to throw from `postinit` methods, enabling unified checking of the instance no matter which initializer was called and allowing the user to respond to the erroneous behavior without halting their program.  However, such support is not currently available.

When this issue was opened, on main it was possible to declare a `postinit` method on a type with `throws` and to throw from that method.  However, when it was defined for classes, the error would stop in the `_new` method, meaning that it could never be caught.  In adding (prototypical) support for throwing from initializers, we will add an explicit check to prevent supporting this feature until we have properly implemented it, such that it can throw errors that are caught and that the memory that would otherwise have been allocated gets cleaned up appropriately.

We don't anticipate adding this support prior to 2.0 at this time.

### Steps to Reproduce

**Source Code:**

```chapel
class Foo {
  var x: int;

  proc postinit() throws {
    if (x > 5) {
      throw new Error("x too large");
    }
  }
}

proc main() {
  try {
    var f1 = new Foo(3);
    writeln(f1);
    var f2 = new Foo(7);
    writeln(f2);
  } catch e {
    writeln(e.message());
  }
}
```

**Compile command:**
`chpl foo.chpl`

**Execution command:**
`./foo`

**Associated Future Test(s):**
[`test/classes/initializers/postInit/throwing.chpl`](https://github.com/chapel-lang/chapel/blob/main/test/classes/initializers/postInit/throwing.chpl) #21424

### Configuration Information

- Output of `chpl --version`: chpl version 1.30.0 pre-release
- Output of `$CHPL_HOME/util/printchplenv --anonymize`: N/A
- Back-end compiler and version, e.g. `gcc --version` or `clang --version`: N/A

**Comment by lydia-duncan** [comment]
@bradcray - does #26430 resolve this issue?

**Comment by bradcray** [comment]
Yep, thanks!  For those who come to this later, note that https://github.com/chapel-lang/chapel/issues/26437 is related and represents additional work required to make throwing postinits behave properly in the presence of fields that require deinitialization.
